Philip-Scott/Notes-up
# Support Multiple Files

## Summary
It's time to support having more than one file to store your pages in Notes-Up! This will have the "disadvantage" of now making the Notes-Up files not invisible anymore for the users, but will allow for easier sync and sharing as well as a stronger separation of your files. 

You will now be able to split your school/work notes from your cooking recipes! 

### Current Status:
![image](https://user-images.githubusercontent.com/10786508/60408281-1f4a0600-9b73-11e9-81df-dc81117d3282.png)

![image](https://user-images.githubusercontent.com/10786508/60763946-f377c600-a033-11e9-8c9f-0d32a98a40ad.png)



## Steps to Test
- Make sure we can still open old files inside the new version
- Make sure that new files are still able to be opened in older versions

## Inspiration 

I really like the way OneNote has the multiple notebook support from within the UI, as well as how it presents what to hide or show from that side panel. 
![image](https://user-images.githubusercontent.com/10786508/59971680-99f0a100-9535-11e9-8b2d-89cefd57dca8.png)

## Known Issues / Things To Do
- [ ] Create and register mime-type
- [x] Open file from file system
- [ ] Migrate, Load and Save the stylesheets in the database
    - Make sure current data is not lost 
- [x] UI to change Notebooks from a list of the recently opened files, as well as just open one from the filesystem - IN PROGRESS
    - With righclick actions to: 
        - [ ] Open file location.
        - [ ] Remove from list.
- [x] Switching notebooks does not clear trash
- [x] UI to pick which side panels we leave opened
    - [x] Pane icons  
    - [x] Headerbar submenu
    - [x] Save state in settings

![image](https://user-images.githubusercontent.com/10786508/59971685-d1f7e400-9535-11e9-9c5f-505ab8adc776.png)

## This PR fixes/implements the following **bugs/features**:

- Fixes #322
- Fixes #211
- Fixes #148
